rust/core-lib/src/file.rs

```diff
@@ -209,6 +229,11 @@ fn try_save(path: &Path, text: &Rope, encoding: CharacterEncoding) -> io::Result
 
     fs::rename(tmp_path, path)?;
 
+    #[cfg(target_family = "unix")]
+    fs::set_permissions(path, Permissions::from_mode(file_info.perms.unwrap()))?;
+
+    filetime::set_file_times(path, file_info.accessed_time.unwrap(), file_info.mod_time.unwrap())?;
```

Comment:
I am not sure if there could be cases where using `unwrap` here might cause a panic? Couldn't cause any while playing around with it, so it might be save.

Reply:
Those timestamps might not be available if the file is on a volume with a specific filesystem I think? Say, NTFS or Samba mounts.

Either way this behavior is wrong, since we obviously did access the file, and modified it as well so both those should be set to the current time. The only thing we have to copy over is the creation timestamp.

Reply:
Is creation time stamp a thing? At least on Linux it doesn't exist

Reply:
> Either way this behavior is wrong, since we obviously did access the file, and modified it as well so both those should be set to the current time.

Oops, right.

Reply:
Windows and macOS appear to have creation time. Ext4 is also said to have it but it's not exposed on linux so not caring about it there is fine by me.

Reply:
What about Fuchsia? Seems like it's also supported by xi

Reply:
Hum, it doesn't seem like there's functionality in Rust/the filetime crate for setting creation time. Since I don't really have a clue of other platforms I'm not quite sure how to go on with this.

Reply:
hmm. This might be rabbit-hole territory (it seems reasonable that filetime would have this functionality.)

Fuchsia we can comfortably ignore for now.

Reply:
Alright, what are we going to do about this then?

Reply:
addressed in review. 🙂